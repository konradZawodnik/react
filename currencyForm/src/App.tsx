import { useCallback, useState } from "react";

import Cash from "./Cash/Cash";
import Select from "./Select/Select";
import Input from "./Input/Input";

import "./App.css";

type ExchangeCounterProps = {
  currencies: Array<{
    id: number;
    name: string;
    ratio: number;
    title: string;
  }>;
  prices: {
    [select: string]: number;
  };
};

type selectedValue = string;

const ExchangeCounter = ({ currencies, prices }: ExchangeCounterProps) => {
  const [amount, setAmount] = useState<string>("");
  const [product, setProduct] = useState<string>("electricity");

  const handleChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
    setAmount(e.target.value);
  }, []);

  const handleSelect = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
    setProduct(e.target.value);
    setAmount("");
  }, []);

  const insertSuffix = useCallback((select: selectedValue) => {
    switch (select) {
      case "electricity":
        return <em>kWh</em>;
      case "petrol":
        return <em>litórw</em>;
      case "oranges":
        return <em>kilogramów</em>;
      default:
        return null;
    }
  }, []);

  const selectPrice = useCallback(
    (select: selectedValue) => {
      const price = prices[select];
      return price;
    },
    [prices]
  );

  const price = selectPrice(product);
  return (
    <div className="App">
      <Select product={product} handleSelect={handleSelect} />
      <br />
      <Input
        amount={amount}
        handleChange={handleChange}
        insertSuffix={insertSuffix}
        product={product}
      />
      {currencies.map((currency) => (
        <Cash
          key={currency.id}
          ratio={currency.ratio}
          title={currency.title}
          cash={amount}
          price={price}
        />
      ))}
    </div>
  );
};

ExchangeCounter.defaultProps = {
  currencies: [
    {
      id: 0,
      name: "złotych",
      ratio: 1,
      title: "Wartość w złotówkach:",
    },
    {
      id: 1,
      name: "dollar",
      ratio: 3.6,
      title: "Wartość w dolarach:",
    },
    {
      id: 2,
      name: "euro",
      ratio: 4.25,
      title: "Wartość w euro:",
    },
    {
      id: 3,
      name: "frank",
      ratio: 3.8,
      title: "Wartość w frankach:",
    },
  ],
  prices: {
    electricity: 0.51,
    petrol: 4.76,
    oranges: 3.79,
  },
};

export default ExchangeCounter;
