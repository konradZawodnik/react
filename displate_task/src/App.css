* {
  margin: 0;
  padding: 0;
}

.App {
  background-color: aquamarine;
  display: grid;
  grid-template-columns: 2fr 2fr 2fr 2fr;
  height: 100%;
  padding-left: 2vw;
  padding-top: 1.5vh;
  width: 100%;
}

.Button {
  border: solid 3px black;
  cursor: pointer;
  font-weight: bold;
  min-width: 50%;
  margin-bottom: 2vh;
  text-transform: capitalize;
}

.Button:hover {
  color: red;
}

.ButtonOfTheSameBreedContainer {
  display: flex;
  flex-flow: wrap;
}

.ButtonOfTheSameBreed {
  border: solid 3px black;
  cursor: pointer;
  font-weight: bold;
  min-width: 50%;
  margin-bottom: 2vh;
  margin-right: 2vh;
  text-transform: capitalize;
}

.ButtonOfTheSameBreed:hover {
  color: red;
}

.Loading {
  display: flex;
  font-weight: bold;
  font-size: 2rem;
  justify-content: center;
}

.Modal {
  position: fixed; 
  padding-top: 100px; 
  right: 0; 
  top: 0;
  bottom: 0; 
  overflow: auto; 
  background-color: rgb(0,0,0); 
  background-color: rgba(0,0,0,0.4); 
  z-index: 9999; 
}

.ModalBody {
  display: flex;
  justify-content: center;
  width: 100%
}

.ModalFooter {
  display: flex;
  justify-content: center;
}

.SiteTitle {
  display: flex;
  background-color: cornflowerblue ;
  font-weight: bold;
  font-size: 2rem;
  justify-content: center;
  height: 100%;
  width: 100%;
}