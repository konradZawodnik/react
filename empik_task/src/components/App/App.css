.container{
    max-width: 1000px;
    margin: 100px auto 0 auto;
    width: 100%;
}

.row {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    padding: 10px;
    border-bottom: 1px solid silver;
    list-style: none;
}

.name {
    font-weight: bold;
}

.price {
    padding-left: 10px;
}

.summary {
    font-weight: bold;
}