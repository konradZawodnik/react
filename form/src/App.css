.App {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.Button {
  cursor: pointer;
}

.Button:hover {
  font-weight: bold;
}
