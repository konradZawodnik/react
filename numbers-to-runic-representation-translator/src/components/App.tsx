import React, { useState, ChangeEvent, FormEvent } from "react";

const runicMapping: { [key: number]: string } = {
  1: "ᚠ",
  2: "ᚡ",
  3: "ᚢ",
  4: "ᚣ",
  5: "ᚤ",
  6: "ᚥ",
  7: "ᚦ",
  8: "ᚧ",
  9: "ᚨ",
  10: "ᚱ",
  20: "ᚲ",
  30: "ᚳ",
  40: "ᚴ",
  50: "ᚵ",
  60: "ᚶ",
  70: "ᚷ",
  80: "ᚸ",
  90: "ᚹ",
  100: "ᚺ",
  200: "ᚻ",
  300: "ᚼ",
  400: "ᚽ",
  500: "ᚾ",
  600: "ᚿ",
  700: "ᛀ",
  800: "ᛁ",
  900: "ᛂ",
  1000: "ᛃ",
  2000: "ᛄ",
  3000: "ᛅ",
  4000: "ᛆ",
  5000: "ᛇ",
  6000: "ᛈ",
  7000: "ᛉ",
  8000: "ᛊ",
  9000: "ᛋ",
};

const generateRunes = (num: number): string => {
  return num
    .toString()
    .split("")
    .map((digit) => runicMapping[parseInt(digit)])
    .join("");
};

const App: React.FC = () => {
  const [number, setNumber] = useState<string>("");
  const [runeOutput, setRuneOutput] = useState<string>("");
  const [svgData, setSvgData] = useState<string>("");

  const handleChange = (event: ChangeEvent<HTMLInputElement>): void => {
    setNumber(event.target.value);
  };

  const handleSubmit = (event: FormEvent): void => {
    event.preventDefault();
    const num = parseInt(number);
    if (num <= 9999 && num > 0) {
      const runicStr = generateRunes(num);
      setRuneOutput(runicStr);
      setSvgData(createSvg(runicStr));
    } else {
      return;
    }
  };

  const createSvg = (runes: string): string => {
    const svg = `
      <svg xmlns="http://www.w3.org/2000/svg" width="500" height="100">
        <text x="10" y="40" font-family="Arial" font-size="30" fill="black">${runes}</text>
      </svg>
    `;
    return svg;
  };

  const handleDownload = (): void => {
    const blob = new Blob([svgData], { type: "image/svg+xml" });
    const link = document.createElement("a");
    link.href = URL.createObjectURL(blob);
    link.download = "runes.svg";
    link.click();
  };

  return (
    <div className="App min-h-screen bg-gray-100 flex items-center justify-center">
      <div className="bg-white p-8 rounded-lg shadow-xl w-96">
        <h1 className="text-2xl font-bold text-center mb-6">
          Decimal to Runic Translator
        </h1>
        <form onSubmit={handleSubmit} className="space-y-4">
          <input
            type="number"
            value={number}
            onChange={handleChange}
            max="9999"
            min="1"
            placeholder="Enter a number"
            className="w-full p-2 border border-gray-300 rounded-md text-lg"
          />
          <button
            type="submit"
            className="w-full p-2 bg-blue-500 text-white rounded-md hover:bg-blue-600"
          >
            Translate
          </button>
        </form>

        {runeOutput && (
          <>
            <h2 className="text-xl font-semibold mt-6">
              Runic Representation:
            </h2>
            <div className="text-center text-3xl mt-2">{runeOutput}</div>

            <h2 className="text-xl font-semibold mt-6">SVG Image:</h2>
            <div className="mt-2">
              <div dangerouslySetInnerHTML={{ __html: svgData }} />
              <button
                onClick={handleDownload}
                className="mt-4 w-full p-2 bg-green-500 text-white rounded-md hover:bg-green-600"
              >
                Download SVG
              </button>
            </div>
          </>
        )}
      </div>
    </div>
  );
};

export default App;
