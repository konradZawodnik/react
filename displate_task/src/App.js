import React, { useCallback, useState, useEffect, Fragment } from 'react';
import axios from 'axios';
import { Button, Modal, ModalBody, ModalFooter } from 'reactstrap';

import './App.css';

const App = () => {
  const [data, setData] = useState([]);
  const [photo, setPhoto] = useState(null);
  const [randomPhoto, setRandomPhoto] = useState(null);
  const [modalOpen, setModalOpen] = useState(false);
  const [loading, setLoading] = useState(false);
  const [active, setActive] = useState(false);
  const getListOfDogs = useCallback(async () => {
    try {
      setLoading(true);
      const response = await axios.get('https://dog.ceo/api/breeds/list/all');
      setData(response.data.message);
      setLoading(false);
    } catch (err) {
      throw err;
    } finally {
      setLoading(false);
    }
  }, []);

  const getImageOfClickedBreed = useCallback(async (e, value) => {
    try {
      setLoading(true);
      setRandomPhoto(null);
      const response = await axios.get(`https://dog.ceo/api/breed/${value.trim()}/images/random`);
      setPhoto(response.data.message);
      setActive(true);
      setModalOpen(true);
      setLoading(false);
    } catch (err) {
      throw err;
    } finally {
      setLoading(false);
    }
  }, []);

  const getImageOfClickedSubBreed = useCallback(async (e, value) => {
    try {
      setLoading(true);
      const response = await axios.get(`https://dog.ceo/api/breed/${value}/images/random`);
      setPhoto(response.data.message);
      setActive(true);
      setModalOpen(true);
      setLoading(false);
    } catch (err) {
      throw err;
    } finally {
      setLoading(false);
    }
  }, []);

  const getRandomImage = useCallback(async () => {
    try {
      setPhoto(null);
      setLoading(true);
      const response = await axios.get('https://dog.ceo/api/breeds/image/random');
      setRandomPhoto(response.data.message);
      setActive(true);
      setModalOpen(true);
      setLoading(false);
    } catch (err) {
      throw err;
    } finally {
      setLoading(false);
    }
  }, []);

  useEffect(() => {
    getListOfDogs();
  }, [getListOfDogs]);

  if (loading) {
    return <div className="Loading">{"Loading"}</div>
  }
  return (
    <Fragment>
      <span className="SiteTitle">{"Breeds of dogs"}</span>
      <div className="App">
        {data && Object.keys(data).map((key, index) => (
          <Fragment key={index}>
            {data[key].length <= 0 &&
              <ul>
                <li key={index}>
                  <Button
                    active={active}
                    className="Button"
                    onClick={(e) => getImageOfClickedBreed(e, `${key}${data[key]}`)}
                  >
                    {`${key} ${data[key]}`}
                  </Button>
                </li>
              </ul>}
            {data[key].length > 0 && (
              <ul>
                <li key={index}>
                  {Object.values(data[key]).map((value, id) => (
                    <div key={id} className="ButtonOfTheSameBreedContainer">
                      <Button
                        active={active}
                        className="ButtonOfTheSameBreed"
                        onClick={(e) => getImageOfClickedSubBreed(e, `${key}`)}
                      >
                        {`${key} ${value}`}
                      </Button>
                    </div>
                  ))}
                </li>
              </ul>)}
          </Fragment>
        ))}
        {active &&
          <div >
            <Modal isOpen={modalOpen} className="Modal">
              <ModalBody className="ModalBody">
                {photo && modalOpen &&
                  <img
                    alt={photo}
                    src={photo}
                  />}
                {randomPhoto && modalOpen &&
                  <img
                    alt={randomPhoto}
                    src={randomPhoto}
                  />}
              </ModalBody>
              <ModalFooter className="ModalFooter">
                <Button
                  className="Button"
                  color="primary"
                  onClick={getRandomImage}
                >
                  {"Losuj nastepne zdjÄ™cie"}
                </Button>
                <Button
                  className="Button"
                  color="secondary"
                  onClick={() => {
                    setPhoto(null);
                    setModalOpen(false);
                  }}
                >
                  {"Zamknij"}
                </Button>
              </ModalFooter>
            </Modal>
          </div>}
      </div >
    </Fragment>
  );
};

export default App;
