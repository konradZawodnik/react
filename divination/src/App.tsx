import { useState, useCallback } from 'react';

import './App.css';

interface PredictionsState {
  [index: number]: { id: number; title: string };
}

let counter = 0;

const App = () => {
  const initialValues: PredictionsState = [{
    id: counter++,
    title: "Pierwsza wróżba"
  },
  {
    id: counter++,
    title: "Druga wróżba"
  },
  {
    id: counter++,
    title: "Trzecia wrózba"
  }]
  const [value, setValue] = useState<string>('');
  const [predictions, setPredictions] = useState<any>(initialValues);

  const handleSubmit = useCallback(() => {
    let randomItem = Math.floor(Math.random() * predictions.length);
    setPredictions(predictions[randomItem]);
  }, [predictions]);

  const handleAddClick = useCallback(() => {
    for (let i = 0; i < predictions.length; i++) {
      const newArray = predictions.slice();
      newArray.push({ id: counter++, title: value });
      if (newArray[newArray.length - 1].title === '') {
        return alert("Napisz jakąś wróżbę");
      } else {
        setPredictions(newArray);
        alert(`Wróżba dodana. Aktulane wrózby to :${newArray[i].title}`)
      };
    };
  }, [predictions, value]);

  return (
    <div className="Container">
      <input
        className="Input"
        type="text"
        onChange={(e) => setValue(e.target.value)}
      />
      <button
        className="Button"
        onClick={handleSubmit}
        type="submit"
      >
        Zobacz wróżbę
       </button>
      <br />
      <button
        className="Button"
        onClick={handleAddClick}
        type="button"
      >
        Dodaj wróżbę
        </button>
      <h1>{predictions.title}</h1>
      <br />
    </div>
  );
}

export default App;
