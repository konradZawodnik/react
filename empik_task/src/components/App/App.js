import axios from 'axios';
import React, { useCallback, useEffect, useState } from 'react';
import { connect } from 'react-redux';

import AddedProductsToBasket from './AddedProductsToBasket/AddedProductsToBasket';

import './App.css';

const App = ({ count }) => {

  const [data, setData] = useState([]);
  const [amount, setAmount] = useState(0);

  const getProducts = useCallback(async () => {
    try {
      const response = await axios.get('/api/cart');
      setData(response.data);
    } catch (err) {
      return err;
    }
  }, [data]);

  useEffect(() => {
    getProducts();
  }, []);

  return (
    <div className="container">
      <h3>Lista produktów</h3>
      <ul>
        {data && data.map((item) => (
          <li className="row" key={item.pid}>
            <div className="name">{item.name}</div>
            <div className="price">{item.price}</div>
            <AddedProductsToBasket
              amount={amount}
              item={item}
              setAmount={setAmount}
            />
          </li>
        ))}
      </ul>
      <div className="summary">{`Całkowita suma zamówienia to ${Number(count * amount).toFixed(2)} zł`}</div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  count: state.countReducer.count,
})

export default connect(mapStateToProps)(App);
