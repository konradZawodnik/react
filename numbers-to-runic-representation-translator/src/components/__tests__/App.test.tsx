import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "../App";

describe("App Component", () => {
  let originalCreateObjectURL: (obj: Blob | MediaSource) => string;

  beforeAll(() => {
    originalCreateObjectURL = URL.createObjectURL;
    URL.createObjectURL = jest.fn(() => "mocked-url");
  });

  afterAll(() => {
    URL.createObjectURL = originalCreateObjectURL;
  });

  it("should match snapshot", () => {
    const counter = render(<App />);
    expect(counter).toMatchSnapshot();
  });

  it("renders correctly", () => {
    render(<App />);
    expect(
      screen.getByText(/Decimal to Runic Translator/i)
    ).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/Enter a number/i)).toBeInTheDocument();
  });

  it("should allow input between 1 and 9999", () => {
    render(<App />);

    const input = screen.getByPlaceholderText(/Enter a number/i);

    fireEvent.change(input, { target: { value: "0" } });
    expect(input).toHaveValue(0);
    expect(input).toBeInvalid();

    fireEvent.change(input, { target: { value: "1" } });
    expect(input).toHaveValue(1);
    expect(input).toBeValid();

    fireEvent.change(input, { target: { value: "10000" } });
    expect(input).toHaveValue(10000);
    expect(input).toBeInvalid();

    fireEvent.change(input, { target: { value: "9999" } });
    expect(input).toHaveValue(9999);
    expect(input).toBeValid();
  });

  it("should trigger the translation on valid input", () => {
    render(<App />);

    const input = screen.getByPlaceholderText(
      "Enter a number"
    ) as HTMLInputElement;
    const button = screen.getByText("Translate");

    fireEvent.change(input, { target: { value: "1234" } });
    fireEvent.click(button);

    const runicOutput = screen.getByText(/Runic Representation:/);
    expect(runicOutput).toBeInTheDocument();

    const downloadButton = screen.getByText("Download SVG");
    expect(downloadButton).toBeInTheDocument();
  });

  it("should generate an SVG based on runic characters", async () => {
    render(<App />);

    const input = screen.getByPlaceholderText("Enter a number");
    const submitButton = screen.getByText("Translate");

    fireEvent.change(input, { target: { value: "456" } });
    fireEvent.click(submitButton);

    await waitFor(() => {
      const svg = document.querySelector("svg");
      expect(svg).toBeInTheDocument();
      expect(svg).toHaveTextContent("ᚣᚤᚥ");
    });
  });

  it("should trigger download when the download button is clicked", async () => {
    render(<App />);

    const input = screen.getByPlaceholderText("Enter a number");
    const submitButton = screen.getByText("Translate");

    fireEvent.change(input, { target: { value: "789" } });
    fireEvent.click(submitButton);

    await waitFor(() => {
      const downloadButton = screen.getByText("Download SVG");
      fireEvent.click(downloadButton);
      expect(global.URL.createObjectURL).toHaveBeenCalled();
    });
  });
});
