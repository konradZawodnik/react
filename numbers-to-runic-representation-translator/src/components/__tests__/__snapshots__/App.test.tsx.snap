// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App min-h-screen bg-gray-100 flex items-center justify-center"
      >
        <div
          class="bg-white p-8 rounded-lg shadow-xl w-96"
        >
          <h1
            class="text-2xl font-bold text-center mb-6"
          >
            Decimal to Runic Translator
          </h1>
          <form
            class="space-y-4"
          >
            <input
              class="w-full p-2 border border-gray-300 rounded-md text-lg"
              max="9999"
              min="1"
              placeholder="Enter a number"
              type="number"
              value=""
            />
            <button
              class="w-full p-2 bg-blue-500 text-white rounded-md hover:bg-blue-600"
              type="submit"
            >
              Translate
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App min-h-screen bg-gray-100 flex items-center justify-center"
    >
      <div
        class="bg-white p-8 rounded-lg shadow-xl w-96"
      >
        <h1
          class="text-2xl font-bold text-center mb-6"
        >
          Decimal to Runic Translator
        </h1>
        <form
          class="space-y-4"
        >
          <input
            class="w-full p-2 border border-gray-300 rounded-md text-lg"
            max="9999"
            min="1"
            placeholder="Enter a number"
            type="number"
            value=""
          />
          <button
            class="w-full p-2 bg-blue-500 text-white rounded-md hover:bg-blue-600"
            type="submit"
          >
            Translate
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
